### PR TITLE
feat: calculate and show sum and average of discipline time spent

### PR DESCRIPTION
**Pull request by Kovalenko77**

### CODE REVIEW

File: src/Tracker.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import TrackerTable from './TrackerTable';
 
+
```
Comment (romanalesenkov1) [inline src/Tracker.jsx]: unneeded changes

File: src/TrackerTable.jsx [hunk 2]
```diff
@@ -1,26 +1,45 @@
...
+import { calculateTotalTimeSpent, calculateAverageTimeSpent } from './utils';
 
 const TrackerTable = ({ items }) => {
     let tableRowDataArr = items.map(function (item) {
         return (
             <tr key={item.id}>
                 <td>{item.date}</td>
-                <td>{item.data.IT.timeSpent}</td>
                 <td>{item.data.english.timeSpent}</td>
+                <td>{item.data.IT.timeSpent}</td>
```
Comment (romanalesenkov1) [inline src/TrackerTable.jsx:10]: why did you change the order of these lines?

File: src/index.css [hunk 3]
```diff
@@ -79,4 +79,10 @@ td {
     border: 1px solid black;
     padding: 8px;
     text-align: center;
+    width: 450px;
 }
+
+.tracker-table-food {
```
Comment (romanalesenkov1) [inline src/index.css]: what kind of food is it?

File: src/index.css [hunk 4]
```diff
@@ -68,7 +68,7 @@ a {
     cursor: pointer;
 }
 
-.table-tracker table {
+.tracker table {
```
Comment (romanalesenkov1) [inline src/index.css]: why do you use this selector?
it will select all tables within .tracker. Is it what you need?

File: src/utils.js [hunk 5]
```diff
@@ -11,3 +11,22 @@ export function getCurrentDateFormatted() {
 
     return formattedDate;
 }
+
+export function calculateTotalTimeSpent(dataArray, discipline) {
+    // Используем метод reduce для свертки массива и вычисления суммы
```
Comment (romanalesenkov1) [inline src/utils.js:16]: it is better to use English

File: src/utils.js [hunk 6]
```diff
@@ -11,3 +11,22 @@ export function getCurrentDateFormatted() {
...
+            // Добавляем значение свойства 'timeSpent' к общей сумме
+            total += currentItem.data[discipline].timeSpent;
+        }
+        return total;
+    }, 0);
+}
+
+export function calculateAverageTimeSpent(dataArray, discipline) {
+    const timeSpentValues = dataArray.map((item) => item.data[discipline].timeSpent);
+    // Вычисляем среднее значение
```
Comment (romanalesenkov1) [inline src/utils.js]: why do you need this comment?

File: src/utils.js [hunk 7]
```diff
@@ -11,3 +11,22 @@ export function getCurrentDateFormatted() {
...
+    const averageTimeSpent = timeSpentValues.length > 0 ? timeSpentValues.reduce((sum, value) => sum + value, 0) / timeSpentValues.length : 0;
```
Comment (romanalesenkov1) [inline src/utils.js]: I would rather use IF statement, it'll be more readable

File: src/Tracker.jsx (hunk 1 above)
Comment (Kovalenko77) [reply src/Tracker.jsx]: Done. Thanks.

File: src/index.css (hunk 3 above)
Comment (Kovalenko77) [reply src/index.css]: Done. Thanks.

File: src/utils.js (hunk 6 above)
Comment (Kovalenko77) [reply src/utils.js]: No, i don`t need. Thanks.

File: src/utils.js (hunk 7 above)
Comment (Kovalenko77) [reply src/utils.js]: Done. Thanks.

File: src/TrackerTable.jsx (hunk 2 above)
Comment (Kovalenko77) [reply src/TrackerTable.jsx:10]: it works as expected

File: src/index.css (hunk 4 above)
Comment (Kovalenko77) [reply src/index.css]: Done.Thanks.

File: src/utils.js [hunk 8]
```diff
@@ -11,3 +11,26 @@ export function getCurrentDateFormatted() {
...
+            // Добавляем значение свойства 'timeSpent' к общей сумме
+            total += currentItem.data[discipline].timeSpent;
+        }
+        return total;
+    }, 0);
+}
+
+export function calculateAverageTimeSpent(dataArray, discipline) {
+    const timeSpentValues = dataArray.map((item) => item.data[discipline].timeSpent);
+    let averageTimeSpent;
```
Comment (romanalesenkov1) [inline src/utils.js]:
```suggestion
    let averageTimeSpent = 0;
    if (timeSpentValues.length > 0) {
        averageTimeSpent = timeSpentValues.reduce((sum, value) => sum + value, 0) / timeSpentValues.length;
    } 
```

Comment (romanalesenkov1) [reply src/utils.js]: why did you just close the comment?

File: src/utils.js [hunk 9]
```diff
@@ -11,3 +11,26 @@ export function getCurrentDateFormatted() {
...
+    if (timeSpentValues.length > 0) {
```
Comment (romanalesenkov1) [inline src/utils.js:30]: refactor it

Comment (Kovalenko77) [reply src/utils.js:30]: done
